kotlinpoet/src/test/java/com/squareup/kotlinpoet/CodeBlockTest.kt

```diff
@@ -604,4 +604,27 @@ class CodeBlockTest {
     // println("""ESCAPE '\'""") -> ESCAPE '\'
     assertThat(CodeBlock.of("%P", """ESCAPE '\'""").toString()).isEqualTo("\"\"\"ESCAPE '\\'\"\"\"")
   }
+
+  // https://github.com/square/kotlinpoet/issues/1381
+  @Test fun useUnderscoresOnLargeDecimalLiterals() {
+    assertThat(CodeBlock.of("%L", 10000).toString()).isEqualTo("10_000")
+    assertThat(CodeBlock.of("%L", 100000L).toString()).isEqualTo("100_000")
+    assertThat(CodeBlock.of("%L", Int.MIN_VALUE).toString()).isEqualTo("-2_147_483_648")
+    assertThat(CodeBlock.of("%L", Int.MAX_VALUE).toString()).isEqualTo("2_147_483_647")
+    assertThat(CodeBlock.of("%L", Long.MIN_VALUE).toString()).isEqualTo("-9_223_372_036_854_775_808")
+    assertThat(CodeBlock.of("%L", 10000.123).toString()).isEqualTo("10_000.123")
+    assertThat(CodeBlock.of("%L", 3.0).toString()).isEqualTo("3.0")
+    assertThat(CodeBlock.of("%L", 10000.123f).toString()).isEqualTo("10_000.123")
+    assertThat(CodeBlock.of("%L", 10000.123456789012).toString()).isEqualTo("10_000.123456789011")
```

Comment:
Need tests for shorts.

Reply:
I added one test for shorts (i.e. `1281.toShort()`) Do you think there are more cases that should be properly tested, that I haven't thought of?